code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1798,3 +1832,104 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 /datum/species/proc/start_wagging_tail(mob/living/carbon/human/H)
 
 /datum/species/proc/stop_wagging_tail(mob/living/carbon/human/H)
+
+///////////////
+//FLIGHT SHIT//
+///////////////
+
+/datum/species/proc/GiveSpeciesFlight(mob/living/carbon/human/H)
+	if(flying_species) //species that already have flying traits should not work with this proc
+		return
+	flying_species = TRUE
+	if(isnull(fly))
+		fly = new
+		fly.Grant(H)
+	if(H.dna.features["wings"] != wings_icon)
+		if(!("wings" in H.dna.species.mutant_bodyparts))
+			H.dna.species.mutant_bodyparts |= "wings"
```

Comment:
The ``"wings" in mutantparts`` check isn't needed. Just do ``mutant_bodyparts |= "wings"`` without the check or use += if you want to keep it.

Also, ``H.dna.species = src``

Reply:
riight. only adds when it's not there. gotta love |=.